src/interaction/interactive_api.jl

```diff
@@ -168,9 +168,107 @@ function select_rectangle(scene; kwargs...)
         end
         return rect_ret
     end
     return rect_ret
 end
+
+"""
+    select_line(scene; kwargs...) -> line
+Interactively select a line (typically an arrow) on a 2D `scene` by clicking the left mouse button,
+dragging and then un-clicking. Return an **observable** whose value corresponds
+to the selected line on the scene. In addition the function
+_plots_ the line on the scene as the user clicks and moves the mouse
+around. When the button is not clicked any more, the plotted line disappears.
+
+The value of the returned line is updated **only** when the user un-clicks.
+
+The `kwargs...` are propagated into `lines!` which plots the selected line.
+"""
+function select_line(scene; kwargs...)
+    key = Mouse.left
+    waspressed = Node(false)
+    line = Node([Point2f0(0,0), Point2f0(1,1)])
+    line_ret = Node([Point2f0(0,0), Point2f0(1,1)])
+    # Create an initially hidden  arrow
+    plotted_line = lines!(
+        scene, line; visible = false, color = RGBAf0(0.1, 0.1, 0.8, 0.5),
+        linewidth = 4, kwargs...,
+    )[end]
+
+    on(events(scene).mousedrag) do drag
+        if ispressed(scene, key) && is_mouseinside(scene)
+            mp = mouseposition(scene)
+            if drag == Mouse.down
+                waspressed[] = true
+                plotted_line[:visible] = true  # start displaying
+                line[][1] = mp
+                line[][2] = mp
+                line[] = line[]
+            elseif drag == Mouse.pressed
+                line[][2] = mp
+                line[] = line[] # actually update observable
+            end
+        else
+            if drag == Mouse.up && waspressed[] # User has selected the rectangle
+                waspressed[] = false
+                line_ret[] = copy(line[])
+            end
+            # always hide if not the right key is pressed
+            plotted_line[:visible] = false
+        end
+        return line_ret
+    end
+    return line_ret
+end
+
+"""
+    select_point(scene; kwargs...) -> point
+Interactively select a point on a 2D `scene` by clicking the left mouse button,
+dragging and then un-clicking. Return an **observable** whose value corresponds
+to the selected point on the scene. In addition the function
+_plots_ the point on the scene as the user clicks and moves the mouse
+around. When the button is not clicked any more, the plotted point disappears.
+
+The value of the returned point is updated **only** when the user un-clicks.
+
+The `kwargs...` are propagated into `scatter!` which plots the selected point.
+"""
+function select_point(scene; kwargs...)
+    key = Mouse.left
+    pmarker = Circle(Point2f0(0, 0), Float32(1))
+    waspressed = Node(false)
+    point = Node([Point2f0(0,0)])
+    point_ret = Node([Point2f0(0,0)])
+    # Create an initially hidden  arrow
+    plotted_point = scatter!(
+        scene, point; visible = false, marker = pmarker, markersize = 20px,
```

Comment:
Markersize should also ideally be themable.

Reply:
once again I don't understand. Please be explicit.

Reply:
Same thing as above - `marker` and `markersize` should be passed as kwargs by the user.

Reply:
What are you talking about? That's not how keyword propagation works, please read my other comment.

Reply:
Ah okay, you saw it.

Reply:
Yep, sorry - forgot to mark as resolved.  I was unaware that kwargs would merge.